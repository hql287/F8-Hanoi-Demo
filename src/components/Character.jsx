// React Libraries
import path from 'path';
import React, {Component} from 'react';

// Svg paths
const CharacterPath = path.resolve(__dirname, './static/imgs/Mario.svg');
const SpeechPath = path.resolve(__dirname, './static/imgs/speech-bubble.gif');

import styled from 'styled-components';

export const Layout = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-end;
  ${props => `
    margin-left: ${props.pos.x}px;
  `};
`;

export const Speech = styled.img`
  margin-right: -300px;
`;

export const Mario = styled.img`
  display: block;
  ${props =>
    props.size &&
    `
    max-width: ${props.size}px;
  `};
  ${props =>
    props.isJumping
      ? `
      margin-bottom: ${props.groundHeight + 100}px;
    `
      : `
      margin-bottom: ${props.groundHeight}px;
    `};
`;

class Character extends Component {
  render() {
    const {pos} = this.props;
    return (
      <Layout {...this.props}>
        <Speech {...this.props} src={SpeechPath} alt="" />
        <Mario
          {...this.props}
          size={60}
          src={CharacterPath}
          alt="Mario Charactor"
        />
      </Layout>
    );
  }
}

export default Character;
