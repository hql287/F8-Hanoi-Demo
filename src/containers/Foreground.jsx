import React from 'react';
import Character from '../components/Character';

import styled from 'styled-components';
export const Layout = styled.div`
  position: relative;
  top: 0;
  left: 0;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  height: 100%;
  width: 100%;
  flex-direction: column;
`;

function Foreground(props) {
  return (
    <Layout>
      <Character {...props} />
    </Layout>
  );
}

export default Foreground;
