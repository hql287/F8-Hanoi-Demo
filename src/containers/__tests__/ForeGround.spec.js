// Libs
import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

// Component to test
import Foreground, { Layout } from '../Foreground';
import Character from '../../components/Character';

// Setup
const pos = { x: 0, y: 0 };
const groundHeight = 160;
const borderHeight = 10;

describe('Renders the Background correctly to the DOM', () => {

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Foreground
        pos={pos}
        showLabels={false}
        groundHeight={groundHeight}
        borderHeight={borderHeight}
      />
    );
  })

  it('should render a Layout component', () => {
    expect(wrapper.find(Layout)).toHaveLength(1);
    expect(wrapper.find(Layout)).not.toHaveLength(2);
  });

  it('should render a Character component', () => {
    expect(wrapper.find(Character)).toHaveLength(1);
    expect(wrapper.find(Character)).not.toHaveLength(2);
  });

  // Test style rules
  it('should render the components with correct style', () => {

  });

  // Snapshot Testing
  it('should match snapShot', () => {
    const tree = renderer
      .create(
        <Foreground
          showLabels={false}
          pos={pos}
          groundHeight={groundHeight}
          borderHeight={borderHeight}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})
