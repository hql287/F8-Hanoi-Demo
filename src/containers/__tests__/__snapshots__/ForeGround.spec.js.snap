// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders the Background correctly to the DOM should match snapShot 1`] = `
<div
  className="Foreground__Layout-ieRzSl bKizyj"
>
  <div
    className="Character__Layout-bdFJdW lbsWfW"
  >
    <img
      alt=""
      className="Character__Speech-eSrpZz eBgjjc"
      src="/Users/quochungle/Dropbox/Sites/Learning/node/YDKJS/mario/src/components/static/imgs/speech-bubble.gif"
    />
    <img
      alt="Mario Charactor"
      className="Character__Mario-kSBfOl bFPqMa"
      size={60}
      src="/Users/quochungle/Dropbox/Sites/Learning/node/YDKJS/mario/src/components/static/imgs/Mario.svg"
    />
  </div>
</div>
`;
